libs/execution_context/ebpf_program.c

```diff
@@ -21,6 +21,23 @@ static size_t _ebpf_program_state_index = MAXUINT64;
 #define EBPF_MAX_HASH_SIZE 128
 #define EBPF_HASH_ALGORITHM L"SHA256"
 
+// The maximum number of iterations to run when executing a BPF_PROG_RUN before yielding to the scheduler. This is to
+// prevent starving other threads on the CPU, but also allows the measurement of the time spent in the BPF program
+// to be reasonably accurate.
+// Assumptions:
+// 1) 2Ghz processor, with worst case of 2 cycles per instruction (taking into account cache misses).
+// 2) Approximately 4 CPU instructions per BPF instruction.
+// 3) Maximum program size of 4K instructions.
+//
+// Therefore:
+// 1) Upper bound on program run time is then around 32K cycles or 16us.
+// 2) Lower bound is around 200ns (cost of the epoch_enter/exit).
+// 3) Program time is then in the range [0.2us,16us] and 64 iterations is then [12.8us, 1000us].
+// 4) Bound on the accuracy of the measurement is then [0.01%-1.5%] depending on the program size, with larger programs
+//    being more accurate.
+// 5) The maximum duration is then 1000us, which is the same as the maximum duration of the scheduler quantum.
+#define EBPF_TEST_RUN_BATCH_SIZE 64
```

Comment:
If I do the above calculations right, you should instead get a batch size of 2 based on the verifier's max instruction count.